### PR TITLE
Add CI container compilation tests for GPU code

### PR DESCRIPTION
**Pull request by ggeorgakoudis**
@koparasy Ready for review

### CODE REVIEW

File: .github/containers/x86_64-broadwell-cuda11.6.1/repo/packages/AMS/package.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# AMSLib Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from spack.package import *
+
+
+class Ams(CMakePackage, CudaPackage):
+
+    homepage = "https://lc.llnl.gov/gitlab/autonomous-multiscale-project/marbl-matprops-miniapp"
```
Comment (koparasy) [inline .github/containers/x86_64-broadwell-cuda11.6.1/repo/packages/AMS/package.py:11]: I believe here you pick our local AMS-package.py. There should be a public one in the spack official repo.

Comment (koparasy) [reply .github/containers/x86_64-broadwell-cuda11.6.1/repo/packages/AMS/package.py:11]: We never do spack install AMS. Which we may need to start considering doing as a integration test. But let us do that in a next iteration.

**Review by koparasy — APPROVED** [review]
Looks good to me. Do you want to address the comment now? Or should we add an issue and fix this in a later iteration.

File: .github/containers/x86_64-broadwell-cuda11.6.1/repo/packages/AMS/package.py (hunk 1 above)
Comment (ggeorgakoudis) [reply .github/containers/x86_64-broadwell-cuda11.6.1/repo/packages/AMS/package.py:11]: Makes sense to do it at a later iteration. I copied whatever we define for our CPU CI

**Pull request merged**